src/wallet/wallet.cpp

```diff
@@ -4126,6 +4126,24 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
     auto locked_chain = chain.lock();
     LOCK(walletInstance->cs_wallet);
 
+    // Unless allowed, ensure wallet files are not reused across chains:
+    if (!gArgs.GetBoolArg("-allowreusewallets", DEFAULT_ALLOWREUSEWALLETS))
+    {
+        WalletBatch batch(*walletInstance->database);
+        CBlockLocator locator;
+        if (batch.ReadBestBlock(locator))
+        {
+            // Wallet is assumed to be from another chain, if its best known
+            // block is not in active chain:
```

Comment:
This comment's described behaviour is definitely wrong: the best known block could be a stale block!

Reply:
Indeed, but this is going away after the @MeshCollider's comments are addressed. I didn't have much time lately, will do it sometime the next two weeks.